packages/core/src/jobs/job-definitions/batchEvaluations/runEvaluationJob.ts

```diff
@@ -1,15 +1,58 @@
-import { env } from '@latitude-data/env'
 import { Job } from 'bullmq'
 
+import { RunErrorCodes } from '../../../constants'
+import { Result } from '../../../lib'
 import { queues } from '../../../queues'
 import {
-  DocumentLogsRepository,
+  DocumentLogsWithErrorsRepository,
   EvaluationsRepository,
 } from '../../../repositories'
+import { isChainError } from '../../../services/chains/ChainStreamConsumer'
 import { runEvaluation } from '../../../services/evaluations/run'
 import { WebsocketClient } from '../../../websockets/workers'
 import { ProgressTracker } from '../../utils/progressTracker'
 
+/**
+ * We only throw an error in an evaluation run in 2 situations
+ *
+ *  1. Is a ChainError of type `unknown`. This is the catch in `runChain` not knowing what happened.
+ *  2. is not a `ChainError` so it means something exploded.
+ */
+function throwIfUnknownError(
+  error: unknown | undefined,
+): asserts error is Error {
+  const isAllGood =
+    !error || (isChainError(error) && error.errorCode !== RunErrorCodes.Unknown)
+
+  if (isAllGood) return
+
+  throw error
+}
+
+async function fetchData({
+  workspaceId,
+  evaluationId,
+  documentLogUuid,
+}: {
+  workspaceId: number
+  evaluationId: number
+  documentLogUuid: string
+}) {
+  const documentLogsScope = new DocumentLogsWithErrorsRepository(workspaceId)
```

Comment:
This is important. I changed the normal `DocumentLogsRepository` for the one with errors.

Some may run an evaluation with a failed document. This way, we can trace the error and show the failed evaluation result.